text/0060-refashion-updates-apis.md

```diff
@@ -0,0 +1,426 @@
+    "enqueuedAt": "2021-08-10T14:29:17.000000Z",
+    "startedAt": "2021-08-10T14:29:18.000000Z",
+    "finishedAt": null
+}
+```
+
+e.g. A fully qualified `task` object in `succeeded` state.
+
+```json
+{
+    "uid": 0,
+    "indexUid": "movies",
+    "status": "succeeded",
+    "type": "settingsUpdate",
+    "details": {
+        "rankingRules": [
+            "typo",
+            "ranking:desc",
+            "words",
+            "proximity",
+            "attribute",
+            "exactness"
+        ]
+    },
+    "duration": "PT1S",
+    "enqueuedAt": "2021-08-10T14:29:17.000000Z",
+    "startedAt": "2021-08-10T14:29:18.000000Z",
+    "finishedAt": "2021-08-10T14:29:19.000000Z"
+}
+```
+
+e.g. A fully qualified `task` object in `failed` state.
+
+```json
+{
+    "uid": 0,
+    "indexUid": "movies",
+    "status": "failed",
+    "type": "settingsUpdate",
+    "details": {
+        "rankingRules": [
+            "typo",
+            "ranking:desc",
+            "words",
+            "proximity",
+            "attribute",
+            "wordsPosition",
+            "exactness"
+        ]
+    },
+    "error": {
+        "message": "invalid criterion wordsPosition",
+        "code": "internal",
+        "type": "internal_error",
+        "link": "https://docs.meilisearch.com/errors#internal",
+    },
+    "duration": "PT1S",
+    "enqueuedAt": "2021-08-10T14:29:17.000000Z",
+    "startedAt": "2021-08-10T14:29:18.000000Z",
+    "finishedAt": "2021-08-10T14:29:19.000000Z"
+}
+```
+
+e.g. A summarized `task` object as a response for a `202 Accepted` HTTP code.
+
+```json
+{
+    "uid": 0,
+    "indexUid": "movies",
+    "status": "enqueued",
+    "enqueuedAt": "2021-08-11T09:25:53.000000Z"
+}
+```
+
+---
+
+#### 5. APIs endpoints
+
+**Get all tasks** | `GET` - `/tasks`
+
+##### Goals
+
+Allows users to list tasks globally regardless of the indexes involved. Particularly useful to visualize all the tasks.
+
+`200` - Response body - `/tasks`
+
+```json
+{
+    "results": [
+        {
+            "uid": 1,
+            "status": "enqueued",
+            "indexUid": "movies_reviews",
+            "type": "documentsAddition",
+            "duration": null,
+            "enqueuedAt": "2021-08-12T10:00:00.000000Z",
+            "startedProcessingAt": null,
+            "finishedAt": null
+        },
+        {
+            "uid": 0,
+            "status": "succeeded",
+            "indexUid": "movies",
+            "type": "documentsAddition",
+            "details": {
+                "numberOfDocuments": 100
+            },
+            "duration": "PT16S",
+            "enqueuedAt": "2021-08-11T09:25:53.000000Z",
+            "startedAt": "2021-08-11T10:03:00.000000Z",
+            "finishedAt": "2021-08-11T10:03:16.000000Z"
+        }
+    ]
+}
+```
+
+##### Requirements
+
+> 💡 `task` objects are contained in a `results` array.
+>
+> 💡 By default, objects are sorted by `desc` order on `uid` field. So the most recent tasks appear first.
```

Comment:
Sorry if this was answered elsewhere, but I'm not sure to understand correctly this sentence.
It appears to me that the two sentences `sorted by desc order on uid field` and `the most recent tasks appear first` are contradictory. 

By returning by `desc order on uid`, what if an index has a higher `uid` than the one we just updated ? 
For example, I have an index `pokemon` that I updated a lot in the past. Now, I'm working on an `movies` index that I just started. If I request the `tasks`, I will first receive the tasks of the `pokemon` index and not the most recent ones (the `movies` one), won't I ? 

Or maybe the `uid` aren't scoped to an index, meaning that the `uid` are unique and I can't have the same `uid` twice for my `pokemon` and `movies` index ?

Sorry if this was explained elsewhere, I'm doing my best to understand this 😄

Reply:
Hello @mdubus! `uid` is the identifier of a task. The identifiers will be unique and sequential in a global way. 

So a first task uid on a new index will start at `lastTaskUid + 1`. 

Thanks for bringing this up, it's not specified. I will clarify this immediately.

Reply:
Thanks a lot ! This is much clearer now ❤️

Reply:
If you are on the `tasks/` route you will have the most recent tasks first regardless of an index.

If you are on the `indexes/pokemon/tasks` route you will the most recent tasks first for the pokemon index.